src/scanners/chromemanifest.js

```diff
@@ -1,32 +1,21 @@
+import BaseScanner from 'scanners/base';
 import ChromeManifestParser from 'parsers/chromemanifest';
 import * as rules from 'rules/chromemanifest';
-import { ignorePrivateFunctions } from 'utils';
 
 
-export default class ChromeManifestScanner {
+export default class ChromeManifestScanner extends BaseScanner {
 
-  constructor(stream, filename) {
-    this.stream = stream;
-    this.filename = filename;
-    this._rulesProcessed = 0;
-  }
+  _defaultRules = rules;
 
-  scan(_ChromeManifestParser=ChromeManifestParser, _rules=rules) {
-    return new Promise((resolve) => {
-      var validatorMessages = [];
-      var cmParser = new _ChromeManifestParser(this.stream, this.filename);
-      return cmParser.parse()
+  _getContents(_ChromeManifestParser=ChromeManifestParser) {
+    return new Promise((resolve, reject) => {
+      var cmParser = new _ChromeManifestParser(this.contents, this.filename);
+      cmParser.parse()
```

Comment:
But if `parse()` blows up that'll implicitly reject the returned promise won't it?